language/move-lang/src/naming/translate.rs

```diff
@@ -387,6 +400,26 @@ fn script(context: &mut Context, escript: E::Script) -> N::Script {
     }
 }
 
+//**************************************************************************************************
+// Friends
+//**************************************************************************************************
+fn friend(context: &mut Context, mident: ModuleIdent, sp!(loc, efriend): E::Friend) -> N::Friend {
+    let nfriend = match efriend {
+        E::Friend_::Ok => {
+            if context.resolve_module(loc, &mident) {
```

Comment:
Remove all this resolve module logic. Aliases need to be fully expanded in expansion

Reply:
I think we might still need this `resolve_module` so we don't get into declaring a non-existing module as a friend (e.g. `friend 0x2::NonExisting`).

This has an implication (which we might want to have though): when compiling module `M`, all its friend declarations need to be passed to the compiler via either `targets` or `deps` in `move_compile(targets, deps, ...)`.